src/libsystemd-network/dhcp-option.c

```diff
@@ -35,6 +36,31 @@ static int option_append(uint8_t options[], size_t size, size_t *offset,
                 *offset += 1;
                 break;
 
+        case SD_DHCP_OPTION_USER_CLASS: {
+                size_t i = 0;
+                char **s;
+
+                STRV_FOREACH(s, (char **) optval)
+                        i += strlen(*s) + 1;
+
+                if (size < *offset + i + 2)
+                        return -ENOBUFS;
+
+                options[*offset] = code;
+                options[*offset + 1] =  i;
+                *offset += 2;
+
+                STRV_FOREACH(s, (char **) optval) {
+                        i = strlen(*s);
```

Comment:
again, there should be a check here, and we should return EINVAL if this is above 255